src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1148,10 +1148,72 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
             case 2:
                 op2 = getArgForHWIntrinsic(sigReader.GetOp2Type(), sigReader.op2ClsHnd);
                 op2 = addRangeCheckIfNeeded(intrinsic, op2, mustExpand, immLowerBound, immUpperBound);
-                op1 = getArgForHWIntrinsic(sigReader.GetOp1Type(), sigReader.op1ClsHnd);
 
-                retNode = isScalar ? gtNewScalarHWIntrinsicNode(retType, op1, op2, intrinsic)
-                                   : gtNewSimdHWIntrinsicNode(retType, op1, op2, intrinsic, simdBaseJitType, simdSize);
+#ifdef TARGET_ARM64
+                if ((intrinsic == NI_AdvSimd_VectorTableLookup) || (intrinsic == NI_AdvSimd_Arm64_VectorTableLookup))
+                {
+                    op1 = impPopStack().val;
+
+                    if (op1->TypeGet() == TYP_STRUCT)
+                    {
+                        assert(op1->OperIsLocal());
+
+                        LclVarDsc* op1VarDsc  = lvaGetDesc(op1->AsLclVar());
+                        unsigned   lclNum     = lvaGetLclNum(op1VarDsc);
+                        unsigned   fieldCount = info.compCompHnd->getClassNumInstanceFields(sigReader.op1ClsHnd);
+
+                        GenTreeFieldList* fieldList = new (this, GT_FIELD_LIST) GenTreeFieldList();
+                        int               offset    = 0;
+                        for (unsigned fieldId = 0; fieldId < fieldCount; fieldId++)
+                        {
+                            GenTreeLclFld* fldNode = gtNewLclFldNode(lclNum, TYP_SIMD16, offset);
+                            fieldList->AddField(this, fldNode, offset, TYP_SIMD16);
+
+                            offset += op1VarDsc->lvSize() / fieldCount;
```

Comment:
```suggestion
                            offset += getTypeSize(TYP_SIMD16);
```
`lvSize` is not terribly cheap.

Reply:
I wanted to generalize this because in future, we could also add APIs for `TYP_SIMD8`. What I can do is extract `p1VarDsc->lvSize() / fieldCount` out of the loop.

Reply:
If we want to generalize this, we would also want to change of the field that's created. So it could be: `offset += genTypeSize(fldNode);`.

Reply:
I would know the `fldNode` type based on `fieldSize` which is `op1VarDsc->lvSize() / fieldCount`. I think I will just use `fieldSize` to forward the `offset`.